rasterio/env.py

```diff
@@ -446,60 +447,60 @@ def wrapper(*args, **kwds):
     return wrapper
 
 
-@attr.s(slots=True)
-@total_ordering
-class GDALVersion:
+class GDALVersion(Version):
     """Convenience class for obtaining GDAL major and minor version components
-    and comparing between versions.  This is highly simplistic and assumes a
-    very normal numbering scheme for versions and ignores everything except
-    the major and minor components."""
+    and comparing between versions. Pre-release version information is ignored."""
 
-    major = attr.ib(default=0, validator=attr.validators.instance_of(int))
-    minor = attr.ib(default=0, validator=attr.validators.instance_of(int))
+    _pattern = re.compile(r'^\d+\.\d+(\.\d+)?')
 
-    def __eq__(self, other):
-        return (self.major, self.minor) == tuple(other.major, other.minor)
+    def __repr__(self):
+        return f"GDALVersion('{self}')"
 
-    def __lt__(self, other):
-        return (self.major, self.minor) < tuple(other.major, other.minor)
+    def __init__(self, version):
```

Comment:
We lose the ability to pass in `major` and `minor` as args/kwargs. Not sure how important that is :man_shrugging: